src/components/table/hooks/useColumnWidths.ts

```diff
@@ -0,0 +1,27 @@
+import { useRef, useCallback } from 'react';
+
+interface UseColumnWidthsReturn {
+  columnWidthsRef: React.MutableRefObject<Map<string, number>>;
+  setCellRef: (columnKey: string) => (element: HTMLDivElement | null) => void;
+}
+
+export const useColumnWidths = (): UseColumnWidthsReturn => {
+  const columnWidthsRef = useRef<Map<string, number>>(new Map());
+
+  const setCellRef = useCallback(
+    (columnKey: string) => (element: HTMLDivElement | null) => {
+      if (element) {
+        const width = element.getBoundingClientRect().width;
+        if (width > 0) {
+          columnWidthsRef.current.set(columnKey, width);
+        }
+      }
```

Comment:
```suggestion
      if (!element) return;
      const width = element.getBoundingClientRect().width;
      if (width > 0) {
        columnWidthsRef.current.set(columnKey, width);
      }
```
Optional: I suggest using `early return pattern` for better code flow and reduced nesting complexity.